compendium/DeclarativeServices/test/gtest/TestServiceReferenceRetrieval.cpp

```diff
@@ -0,0 +1,91 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "TestFixture.hpp"
+#include "cppmicroservices/Constants.h"
+#include "cppmicroservices/ServiceObjects.h"
+#include "cppmicroservices/ServiceReference.h"
+#include "gtest/gtest.h"
+
+#include "TestInterfaces/Interfaces.hpp"
+
+namespace test
+{
+    struct int1
+    {
+        [[nodiscard]] virtual int getValue() const = 0;
+        virtual ~int1() = default;
+        int1(int1 const&) = default;
+        int1& operator=(int1 const&) = default;
+        int1(int1&&) noexcept = default;
+        int1& operator=(int1&&) noexcept = default;
+
+      protected:
+        int1() = default;
+    };
+
+    struct TestServiceA : public int1
+    {
+        TestServiceA(int val) : id(val) {}
+        [[nodiscard]] int
+        getValue() const override
+        {
+            return id;
+        }
+
+      private:
+        int id;
+    };
+    /*
+     * Tests that a user is able to retrieve a serviceReference from a serviceObject from the coreFramework or DS
+     */
+    TEST_F(tServiceComponent, testServiceReferenceRetrievalForDSAndManReg)
```

Comment:
this is a good set of testpoints. I think there is one case missing; what happens if you:
1. register a service, say `foo`
2. get a service ref to `foo`
3. get an instance of the service object `foo`
4. call `cppmicroservices::ServiceReferenceFromService` passing in the `foo` from step 3
5. unregister `foo` with the cppms framework

What is the expected state of the service ref returned by `cppmicroservices::ServiceReferenceFromService`? I would expect it to act the same as any other service ref that had its service unregistered with the cppms framework.